pytrip/cube.py

```diff
@@ -630,25 +630,42 @@ def parse_path(cls, path_name):
         # 4. file with unknown extensions (i.e. blabla.mp3)
         # Options 1-3 can be parsed, option 4 will give us empty result
 
-        if uncompressed_path_name.endswith(class_header_file_extension):  # header file
+        if uncompressed_path_name.endswith(class_header_file_bare_extension):  # path to header file
             logger.debug("header file ")
-            header_file = uncompressed_path_name
-            basename_end_index = uncompressed_path_name.rfind(class_header_file_extension)
-            cube_filename = uncompressed_path_name[:basename_end_index] + class_data_file_extension
-        elif uncompressed_path_name.endswith(class_data_file_extension):  # cube file
+            basename_end_index = uncompressed_path_name.rfind(class_header_file_bare_extension)
+            basename = uncompressed_path_name[:basename_end_index]
+            if basename[-1] == '.':
+                basename = basename[:-1]
+        elif uncompressed_path_name.endswith(class_data_file_bare_extension):  # cube file
             logger.debug("cube file ")
-            basename_end_index = uncompressed_path_name.rfind(class_data_file_extension)
-            header_file = uncompressed_path_name[:basename_end_index] + class_header_file_extension
-            cube_filename = uncompressed_path_name
+            basename_end_index = uncompressed_path_name.rfind(class_data_file_bare_extension)
+            basename = uncompressed_path_name[:basename_end_index]
+            if basename[-1] == '.':
+                basename = basename[:-1]
         elif not os.path.splitext(uncompressed_path_name)[1]:  # file without extension
             logger.debug("file without extension ")
-            header_file = uncompressed_path_name + class_header_file_extension
-            cube_filename = uncompressed_path_name + class_data_file_extension
-        else:   # a problem (i.e. uknown exception)
+            basename = uncompressed_path_name
+        else:   # a problem (i.e. unknown exception)
             logger.debug("a problem (unknown extension?) with path " + path_name)
-            header_file, cube_filename = None, None
+            basename = None
 
-        return header_file, cube_filename
+        return basename, class_header_file_bare_extension, class_data_file_bare_extension
+
+    @classmethod
```

Comment:
It doesn't require object of a class to call it.
You can call it via `CtxCube.header_file_name(...)` or `DosCube.header_file_name(...)`

 It is however different from `@staticmethod`. It takes `cls` parameter which is a class on which it is called.
So it can access class-level parameters (such as file extensions).